cmd/govim/quickfix.go

```diff
@@ -24,32 +28,29 @@ func (q quickfixEntry) equalModuloBuffer(q2 quickfixEntry) bool {
 }
 
 func (v *vimstate) quickfixDiagnostics(flags govim.CommandFlags, args ...string) error {
-	wasNotDiagnostics := !v.quickfixIsDiagnostics
-	v.quickfixIsDiagnostics = true
-	return v.updateQuickfixWithDiagnostics(true, wasNotDiagnostics)
+	return v.updateQuickfixWithDiagnostics(true)
 }
 
 // updateQuickfixWithDiagnostics updates Vim's quickfix window with the current
-// diagnostics(), respecting config settings that are overridden by force.  The
-// rather clumsily named wasPrevNotDiagnostics can be set to true if it is
-// known to the caller that the quickfix window was previously (to this call)
-// being used for something other than diagnostics, e.g. references.
-func (v *vimstate) updateQuickfixWithDiagnostics(force bool, wasPrevNotDiagnostics bool) error {
+// diagnostics(), respecting config settings that are overridden by force.
+func (v *vimstate) updateQuickfixWithDiagnostics(force bool) error {
 	if !force && (v.config.QuickfixAutoDiagnostics == nil || !*v.config.QuickfixAutoDiagnostics) {
 		return nil
 	}
-	diagsRef := v.diagnostics()
-	work := v.lastDiagnosticsQuickfix != diagsRef
-	v.lastDiagnosticsQuickfix = diagsRef
-	if (!force && !work) || !v.quickfixIsDiagnostics {
+
+	var (
```

Comment:
Please change this from a `var` block to short variable declarations.

Reply:
@myitcv sure no problem, by curiosity why do you prefer this kind of declaration ? On my side I like to pack variable declarations like that, because that formats the code and I found it's more clear and readable.

Reply:
I've honestly not seen much (any?) use of parenthesised `var` declaration lists "outside" of top level file declarations. At the file level they do have things to stand out more, and when combined with `import` and `const` declarations, visually make sense. But within a function, they're a bit clumsy and unnecessarily verbose to my mind: which is precisely what short variable declarations exist to help with (amongst other situations like `for` loops).

Reply:
Ok, as you wish b65554c